src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Protocol/AutoInsert/RemoteInsertTextEdit.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using static Microsoft.VisualStudio.LanguageServer.Protocol.VsLspExtensions;
+using static Roslyn.LanguageServer.Protocol.RoslynLspExtensions;
+using RoslynInsertTextFormat = Roslyn.LanguageServer.Protocol.InsertTextFormat;
+
+namespace Microsoft.CodeAnalysis.Razor.Protocol.AutoInsert;
+
+[DataContract]
+internal readonly record struct RemoteInsertTextEdit(
+        [property: DataMember(Order = 0)]
+        LinePositionSpan LinePositionSpan,
+        [property: DataMember(Order = 1)]
+        string NewText,
+        [property: DataMember(Order = 2)]
+        RoslynInsertTextFormat InsertTextFormat
+    )
+{
+    public static RemoteInsertTextEdit FromLspInsertTextEdit(VSInternalDocumentOnAutoInsertResponseItem edit)
+        => new (
+            edit.TextEdit.Range.ToLinePositionSpan(),
+            edit.TextEdit.NewText,
+            (RoslynInsertTextFormat)edit.TextEditFormat);
+
+    public static VSInternalDocumentOnAutoInsertResponseItem ToLspInsertTextEdit(RemoteInsertTextEdit edit)
+        => new()
+        {
+            TextEdit = new()
+            {
+                Range = VsLspExtensions.ToRange(edit.LinePositionSpan),
+                NewText = edit.NewText
+            },
```

Comment:
Please use `VsLspFactory.CreateTextEdit(...)`, which has a lot of overloads. I think you can just write this:

```suggestion
            TextEdit = VsLspFactory.CreateTextEdit(edit.LinePositionSpan), edit.NewText),
```